### DISCUSSION TITLE
Upstream Galaxy now uses Galaxy NG, does not support API v2

### DISCUSSION THREAD

**Discussion by jcox10** [briantist/galactory#102] [Announcements]
Looks like Galaxy is testing out some new version, and I noticed that the API V2 is now missing?

https://galaxy.ansible.com/api/

```
HTTP 200 OK
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "available_versions": {
        "v3": "v3/",
        "pulp-v3": "pulp/api/v3/",
        "v1": "v1/"
    },
    "server_version": "4.9.0dev",
    "galaxy_ng_version": "4.9.0dev",
    "galaxy_ng_commit": "master:9f4ddc6c",
    "galaxy_importer_version": "0.4.13",
    "pulp_core_version": "3.28.12",
    "pulp_ansible_version": "0.19.0",
    "pulp_container_version": "2.15.2"
}
```

**Comment by jcox10** [comment]
I guess we can get old galaxy at https://old-galaxy.ansible.com ... not having a warm fuzzy feeling about their new galaxy rollout 🙄 The whole point of api versions is to create a new one, not get rid of the old ones LOL

**Reply by jcox10** [reply]
I posted this question https://forum.ansible.com/t/api-v2-missing-in-galaxy-ng/1406

**Comment by briantist** [comment]
Hi @jcox10 ,

They aren't doing testing, this is a permanent change to public Galaxy to use [galaxy_ng](https://github.com/ansible/galaxy_ng). The cutover was made this past weekend.

I was aware of the change, but I hadn't realized that NG no longer supported `v2`; so it caught me by surprise as well.

In my own environments I've switched galactory to use `old-galaxy.ansible.com` as its upstream. Do note that old galaxy is not intended to stick around forever, and it's supposed to be a frozen read-only version of the state of galaxy when the cutover was done, so it is not a long-term solution.

See also:
- https://github.com/briantist/galactory/issues/23#issuecomment-1740003748

I'm currently crunching on adding `v3` support to galactory; progress is not pushed up to a PR yet, but expect to see something soon.

---

Another possible solution is to turn off proxying in galactory, and use multiple servers configured in `ansible.cfg`. That allows `ansible-galaxy` to fall back, but I don't know off-hand all of the situations where the CLI will fall back vs. fail (like for dependencies and things like that). I don't quite have the time to investigate it either and will focus on getting v3 support out the door.

**Reply by briantist** [reply]
Also because it might be useful to anyone reading this, here's a tracker of issues that the RedHat galaxy team is tracking:
- https://issues.redhat.com/issues/?filter=12422312

**Reply by briantist** [reply]
- #104

**Comment by briantist** [comment]
`v0.11.1` is out and ready, and it includes support for galaxy v3 as a server and a client (upstream proxying).

- https://github.com/briantist/galactory/blob/main/CHANGELOG.rst#v0-11-1
- https://pypi.org/project/galactory/0.11.1/
- https://github.com/briantist/galactory/releases/tag/v0.11.1
- https://github.com/briantist/galactory/pkgs/container/galactory/135421376?tag=0.11.1
